ouroboros-consensus/docs/internal-CHANGELOG.md

```diff
@@ -0,0 +1,83 @@
+# Internal Change Log
+
+  - This change log is primarily intended to benefit downstream Cardano
+    development teams. We therefore are not commiting to discuss changes that
+    aren't relevant to them -- eg those changes would likely be most useful to
+    list in the `cardano-node` release notes.
+
+  - So far, versioning of "internal" Cardano libraries is based only on commit
+    hashes, except for the `cardano-node` repo.
+
+For those reasons, we primarily organize this log chronologically: most recent
+changes first.
+
+In order to identify changes by commit hash, Pull Request number, precise
+timestamp, etc, invoke `git blame --first-parent` or `git blame` on this file.
+
+Relatedly, we will NEVER alter entries in this log (until we shift to an
+entirely different scheme). Entries (ie the lines themselves) in this log are
+immutable once their PR is merged into the mainline branch. If there is an
+extremely problematic typo eg, then I suggest pseudo-violating the immutability
+rule only by _adding_ a new corrected line nearby. Ideally such new lines are
+surrounded by lines from the original commit -- this will make the `git blame`
+output easiest to parse.
+
+Each entry is organized similar to https://keepachangelog.com/en/1.1.0/, adapted
+to our plan explained above.
+
+## 2021 Jul 28
```

Comment:
So in this case we won't be able to retrieve the commits related to the changes below based on this changelog blame. Is this an exception worth mentioning?

Reply:
I think the new text I just wrote does a better job of that than the old "bootstrapping" word I was hoping would cover it all by itself.

Please let me know if that's clearer now.

Reply:
If you refer to this line:

> Now that this file exists,
>  entries added to it by later PRs will be narrower than this entry. This entry
>  should be the only one that is bootstrapping the file by discussing multiple
>  recent changes.

then yeah, it is clearer now. Thanks.